core/src/com/unciv/logic/city/City.kt

```diff
@@ -206,6 +210,15 @@ class City : IsPartOfGameInfoSerialization, INamed {
 
     fun getGreatPersonPercentageBonus() = GreatPersonPointsBreakdown.getGreatPersonPercentageBonus(this)
     fun getGreatPersonPoints() = GreatPersonPointsBreakdown(this).sum()
+    
+    fun gainStockpiledResource(resourceName: String, amount: Int) {
+        resourceStockpiles.add(resourceName, amount)
+    }
+
+    fun gainStockpiledResource(resource: TileResource, amount: Int) {
+        if (resource.isCityWide) resourceStockpiles.add(resource.name, amount)
+        else civ.gainStockpiledResource(resource.name, amount)
+    }
```

Comment:
Very weird that in the first function you don't check if it's citywide or not, you get 2 different behaviours for civwide resources in the 2 identically named functions